src/uu/ls/src/ls.rs

```diff
@@ -1092,6 +1082,8 @@ fn list(locs: Vec<String>, config: Config) -> i32 {
     let mut dirs = Vec::<PathData>::new();
     let mut has_failed = false;
 
+    let mut out = BufWriter::new(stdout());
```

Comment:
Please see if we can use something like in #2111 where we only buffer if stdout is not interactive (instead of buffering always).
Always buffering will slightly impact user experience when looking at output on a terminal as nothing will be printed until buffer is full.

Might I also suggest keeping this as a global because it might unnecessarily complicate function signatures (although globals are generally frowned upon, this seems like valid usecase to me)

Reply:
I figured we could just always buffer because the output of `ls` should always be generated fairly quickly. At least when I tried there seemed to be no real visible difference. The only time this might be an issue is with an `--ignore` flag that hides a lot of files? One thing I didn't like about that PR was that it needed `Box<dyn Write>`. Maybe it'd be nice to create a `StdoutWriter` type in `uucore` that automatically buffers if's not interactive?

I chose not to make this a global, because `write!` requires a mutable reference, so it would needs in a `Mutex` if we use `lazy_static`, which also complicates the code. Anyway, if you can find a nicer way to implement this, feel free to do so!

Reply:
Agreed, I also noticed https://github.com/rust-lang/rust/issues/60673 - seems like stdout buffering is already in the works.
As for the global related comment - yeah seems like the complicated lazy_static + mutex would be the way to implement it, I'll check if I can find a better way.

Reply:
That issue is very interesting, I hope some solution lands soon! Thanks for checking out the global variable!